website/docs/reference/api/legacy/unleash/admin/features-v2.md

```diff
@@ -740,6 +740,137 @@ Authorization:$KEY
 }
 ```
 
+### Put variants for Feature Toggle on environment {#update-variants-per-environment}
+
+<ApiRequest verb="put" url="api/admin/projects/:projectId/features/:featureName/environments/:environment/variants" title="Create (overwrite) variants for a feature toggle under the provided environment (example data)" payload={[
+	{
+		"name": "variant1",
+		"weightType": "fix",
+		"weight": 650,
+        "payload": {
+          "type": "json",
+          "value": "{\"key1\": \"value\", \"key2\": 123}"
+        },
+        "stickiness": "userId",
+        "overrides": [ {
+          "contextName": "userId",
+          "values": ["1", "23"]
+        } ]
+	},
+	{
+		"name": "variant2",
+		"weightType": "variable",
+		"weight": 123
+	}
+]}/>
+
+This overwrites the current variants for the feature toggle specified in the :featureName parameter for the :environment parameter.
+The backend will validate the input for the following invariants
+
+* If there are variants, there needs to be at least one variant with `weightType: variable`
+* The sum of the weights of variants with `weightType: fix` must be below 1000 (< 1000)
```

Comment:
Is this correct? It's < and not <= 1000, right?

```suggestion
* The sum of the weights of variants with `weightType: fix` must be strictly less than 1000 (< 1000)
```

Reply:
Actually, we require to have at least one variable variant, and the sum of the fixed ones can be 1000 (leaving the variable variant at zero). So we should change the doc, but at the same time I'm thinking that putting this behaviour here might lead to inconsistencies if we change it later and makes me wonder about the level of detail we want to have because these nuances could be in the OpenAPI doc

Reply:
Got it, so it should be something like this instead, then:

```suggestion
* The sum of the weights of variants with `weightType: fix` must be less than or equal to 1000 (≤ 1000)
```

> So we should change the doc, but at the same time I'm thinking that putting this behaviour here might lead to inconsistencies if we change it later and makes me wonder about the level of detail we want to have because these nuances could be in the OpenAPI doc

I'm not entirely sure what you're saying here; could you explain a bit? Anyway, I think this should absolutely go in the OpenAPI doc. This is behavior that people using the endpoints will want to know about. And if it's something that we do end up changing later, then we'll need to update the API docs too.

Reply:
So, the thing is to add a fixed variant, you first need to add a variable variant. This variable variant needs to have at least weight of 1. So the fixed variant can weight at most 999, but never 1000.

Reply:
We could say here something like:
> The sum of the weights of variants with `weightType: fix` must be less than or equal to 1000 (which should include at least one variant with `weightType: variable`)